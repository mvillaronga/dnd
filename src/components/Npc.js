import React from 'react'
import PropTypes from 'prop-types'

const Npc = ({ onClick, name }) => (
  <li
    onClick={onClick}
  >
    {name}
  </li>
)

Npc.propTypes = {
  onClick: PropTypes.func.isRequired,
  name: PropTypes.string.isRequired
}

export default Npc