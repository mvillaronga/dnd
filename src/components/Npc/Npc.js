import React from 'react'
import PropTypes from 'prop-types'

const Npc = ({ onClick, name, selected }) => (
  <li
    style={{
      color: selected ? 'red' : 'blue'
    }}
    onClick={onClick}
  >
    {name}
  </li>
)

Npc.propTypes = {
  onClick: PropTypes.func.isRequired,
  name: PropTypes.string.isRequired
}

export default Npc