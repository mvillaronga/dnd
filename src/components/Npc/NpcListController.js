import React from 'react'
import { useDispatch, useSelector } from 'react-redux'

import {toggleNPC} from '../../redux/actions'

import NpcList from './NpcList'

const NpcListController = () => {
  const dispatch = useDispatch()

  const props = {
    npcs: useSelector(state => state.npcs),

    onNpcClick: (id) => {
      dispatch(toggleNPC(id))
    }
  }

  return <NpcList {...props} />
}

export default NpcListController