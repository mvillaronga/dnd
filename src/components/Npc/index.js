import React from 'react'
import { useDispatch, useSelector } from 'react-redux'

import { toggleNpc } from '../../redux/npcs'

import NpcList from './NpcList'

const NpcListController = () => {
  const dispatch = useDispatch()

  const props = {
    npcs: useSelector(state => state.npcs),

    onNpcClick: (id) => {
      dispatch(toggleNpc(id))
    }
  }

  return <NpcList {...props} />
}

export default NpcListController