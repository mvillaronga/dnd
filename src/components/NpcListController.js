import { connect } from 'react-redux'
// import { toggleTodo } from '../redux/actions'
import NpcList from '../components/NpcList'

// const getVisibleTodos = (todos, filter) => {
//   switch (filter) {
//     case 'SHOW_ALL':
//       return todos
//     case 'SHOW_COMPLETED':
//       return todos.filter(t => t.completed)
//     case 'SHOW_ACTIVE':
//       return todos.filter(t => !t.completed)
//   }
// }

const mapStateToProps = state => {
  return {
    // todos: getVisibleTodos(state.todos, state.visibilityFilter)
    npcs: state.npcs
  }
}

const mapDispatchToProps = dispatch => {
  return {
    onNpcClick: id => {
    //   dispatch(toggleTodo(id))
    
    }
  }
}

const NpcListController = connect(
  mapStateToProps,
  mapDispatchToProps
)(NpcList)

export default NpcListController