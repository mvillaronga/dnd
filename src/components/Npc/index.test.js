import React from 'react'
import thunk from 'redux-thunk'
import configureMockStore from 'redux-mock-store'
import { Provider } from 'react-redux'
import Adapter from 'enzyme-adapter-react-16';
import { configure, mount } from 'enzyme';

import NpcListController from '../Npc'
import Npc from './Npc'

configure({ adapter: new Adapter() });

let state = {
  npcs: [
   {id:0, name:'gandalf', selected: false},
   {id:1, name:'frodo', selected: true}
 ]
 }
describe('NpcListController tests', () => {
   const mockStore = configureMockStore([thunk])
   const store = mockStore(state)

  beforeEach(() => {
    store.clearActions();
  })

  it('should render the panel', () => {

    const comp = mount(
      <Provider store={store}>
        <NpcListController />
      </Provider>
    )
    expect(comp.html()).toMatchSnapshot()
  })

  it('should toggle npc color on selected', () => {
    const comp = mount(
      <Provider store={store}>
        <NpcListController />
      </Provider>
    )

    comp.find(Npc).first().simulate('click')
    expect(store.getActions()).toEqual([ { type: 'TOGGLE_NPC', id: 0 } ])
  })
})