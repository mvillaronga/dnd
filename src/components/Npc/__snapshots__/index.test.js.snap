// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NpcListController tests should render the panel 1`] = `"<ul><li style=\\"color: blue;\\">gandalf</li><li style=\\"color: red;\\">frodo</li></ul>"`;
